numpy/ma/core.py

```diff
@@ -6397,6 +6397,9 @@ def __repr__(self):
             # it's a subclass, or something is wrong, make it obvious
             return object.__repr__(self)
 
+    def __format__(self, format_spec):
+        return str(masked_print_option._display)
```

Comment:
Should this reject some format spec's (or the other way around, only accept certain ones)?

Reply:
Hmm. That's a good question. I'll dig around and see if I can figure out how, perhaps nan is treated, as a model for this. Not sure I'm in a position to implement the full https://www.python.org/dev/peps/pep-3101/#standard-format-specifiers spec.

Reply:
Well, we could start off a bit limited possibly. Maybe some of it can be handled by formatting twice (not quite sure). But if the user requests a certain alignment, it would not be nice to just ignore that either.

Reply:
One option would be `format(str(masked_print_option._display), format_spec)`

Reply:
Yeah, I think I like that option. There is a bit of a difficulty around whether or not to support `f` or `d`, but basically going the string way is a good safe bet.

Only issue is that the error would read:
```
ValueError: Unknown format code 'd' for object of type 'str'
```
I suppose we could replace that with our own error.

Reply:
Great suggestion, that looks correct and (I missed earlier) what's suggested in the PEP https://www.python.org/dev/peps/pep-3101/#controlling-formatting-on-a-per-type-basis

Reply:
Actually, this seems important since this will be used in a context where users are probably expecting a integer or floating type. Thoughts on slicing last char from `format_string` if it ends in a `d` or `f`?

Reply:
That's a good point. Perhaps we should not open this can of worms in this PR, and just require for now that the spec is the empty string.

There's an open issue about implementing `__format__` in numpy.

Reply:
Hmm, unfortunately, it seems partway open now... But to make some tradoffs explicit: 

- Doing nothing else now gives misleading error message (user thought they were formatting a float)
- Empty format string sacrifices ability to align (check if empty, or impose empty?)
- Introduce conditional to slice format type, but may have performance impact (how important is performance for string formatting?)

However, I think any of these cases are better than the current behaviour printing "0.0".

Reply:
And a fourth option, I believe what @seberg was suggesting,

- except and raise a better exception. Perhaps something like "Masked arr marker is str type, omit presentation type when formatting masked array"

Reply:
Yeah, although getting the error details right is maybe not trivial. I suppose you could chain exceptions, but I am not sure it actually helps a lot, I sometimes feel the typical scientist "new to programming" user already needs persuasion to parse through a single traceback...